pegasus-common/src/main/java/com/linkedin/common/callback/Callback.java

```diff
@@ -32,4 +35,13 @@
    * @param e the error
    */
   void onError(Throwable e);
+
+  /**
+   * Called with lazy Throwable creation if the asynchronous operation failed with an error
+   * @param throwableSupplier the supplier to create Throwable when needed
+   */
+  default void onError(Supplier<Throwable> throwableSupplier)
+  {
+    onError(throwableSupplier.get());
+  }
```

Comment:
Recommend to avoid this change to a simple interface. I think the suggestion below for calling supplier.get() can help us avoid the need of this change.

Reply:
See my earlier comments.